### PR TITLE
add CITATION.cff and lint workflow

### LINKED ISSUE #567: add CITATION.cff file
see https://citation-file-format.github.io/

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
add `CITATION.cff`

## Related Issues

Closes #567 

## Checks

- [x] update manifest file

### CODE REVIEW

**Comment by CagtayFabry** [comment]
please feel free to update your ORCID or other metadata @vhirtham @marscher

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  -
+    affiliation: "Bundesanstalt für Materialforschung und -prüfung (BAM)"
+    family-names: Fabry
+    given-names: Cagtay
+    orcid: "https://orcid.org/0000-0002-0788-8079"
+  -
+    affiliation: "Bundesanstalt für Materialforschung und -prüfung (BAM)"
+    family-names: Hirthammer
+    given-names: Volker
+  -
```
Comment (vhirtham) [inline CITATION.cff:16]:
```suggestion
    given-names: Volker
    orcid: " https://orcid.org/0000-0002-0160-969X"
  -
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    orcid: "https://orcid.org/0000-0002-7983-4387"
+  -
+    affiliation: "Bundesanstalt für Materialforschung und -prüfung (BAM)"
+    family-names: Pittner
+    given-names: Andreas
+    orcid: "https://orcid.org/0000-0002-5019-7775"
+cff-version: "1.1.0"
+message: "If you use this software, please cite it as below."
+repository-code: "https://github.com/BAMWelDX/weldx"
+...
```
Comment (marscher) [inline CITATION.cff:30]: should this not point to an article instead of the repo?

Comment (CagtayFabry) [reply CITATION.cff:30]: yes should be updated to point to the latest publication once available

**Comment by CagtayFabry** [comment]
we just need to make sure to fill out/update these fields before tagging a release

```yaml
version:
date-released:
```

**Comment by marscher** [comment]
the release workflow is currently documented in a github discussion, right? Otherwise I'd update that guide document in the repo.

**Comment by CagtayFabry** [comment]
> the release workflow is currently documented in a github discussion, right? Otherwise I'd update that guide document in the repo.

[it was in discussions](https://github.com/BAMWelDX/weldx/discussions/424) but it is probably easier to spot in the repo right? (also to follow updates)
see https://github.com/BAMWelDX/weldx/pull/570

**Comment by vhirtham** [comment]
> > the release workflow is currently documented in a github discussion, right? Otherwise I'd update that guide document in the repo.
> 
> [it was in discussions](https://github.com/BAMWelDX/weldx/discussions/424) but it is probably easier to spot in the repo right? (also to follow updates) see #570

As an alternative or addition, we could also just create a pull request template for releases. As far as I can remember this is possible, but you need to know that it exists and you have to manually update the url when creating a new PR

**Pull request merged**